### PR TITLE
Add support of all Jenkins jobs

### PR DESCRIPTION
**Pull request by j-mracek**
Important: New syntax of --dnf_docke_test introduced.

### CODE REVIEW

**Comment by j-mracek** [comment]
This is overview of new features in this pull-request:
* Remove test-launcher.py file
* Enable dnf-docker-tests for all jobs on jenkins.
* Modify When step where full command or dnf command (dnf-2, dnf-3) can be used.
* Direct comparison between rpm -qa and dnf repoquery --installed
* Add new steps for install remove test, where dnf install is used to install specific version and downgrade and upgrade to specific version.
* Add new steps in install_remove to test dnf install provide, local file, URL, and  dnf remove provide.
* Add  new steps in install_remove to test dnf install @group and dnf group install group and dnf group install group with-options , and  dnf remove @group and dnf group remove group, and option where mandatory and optional package were installed before group install.
* Add additional steps in dowgrade test including --allowerasing option
* Add direct comparison of rpm and dnf lists of installed packages
* Add test - dnf install /path/to/files/*.rpm  
* Add support for check output from commands
*Add test for group list command if correct output

File: dnf-docker-test/features/install_remove-1.feature [hunk 1]
```diff
@@ -32,3 +32,80 @@ Scenario: Install TestA from repository "test-1"
...
+ Then package "TestB" should be "installed"
+ When I execute "dnf" command "remove -y TestB" with "success"
+ Then package "TestB" should be "removed"
+
+ When I execute "dnf" command "install -y /var/www/html/repo/test-1/TestB-1.0.0-1.noarch.rpm" with "success"
+ Then package "TestB" should be "installed"
+ When I execute "dnf" command "remove -y TestB" with "success"
+ Then package "TestB" should be "removed"
+
+ When I execute "command line" command "mkdir /test" with "success"
```
Comment (jsilhan) [inline dnf-docker-test/features/install_remove-1.feature]: `"command line" command` seems clumsy to me. what about `"bash" command`?

File: dnf-docker-test/features/install_remove-1.feature [hunk 2]
```diff
@@ -32,3 +32,80 @@ Scenario: Install TestA from repository "test-1"
...
+ When I execute "command line" command "cp /var/www/html/repo/test-1/Test{A,B,C}*.rpm /test" with "success"
+ When I execute "dnf" command "install -y /test/*.rpm" with "success"
+ Then package "TestA, TestB, TestC" should be "installed"
+ When I execute "dnf" command "remove -y TestA TestB TestC" with "success"
+ Then package "TestA, TestB, TestC" should be "removed"
+
```
Comment (jsilhan) [inline dnf-docker-test/features/install_remove-1.feature]: We should check that before the group was not installed. i.e.:
```python
When I execute "dnf" command "group list Testgroup" with "success"
Then output line should "start" with "Availbale groups:"
```

File: dnf-docker-test/features/install_remove-1.feature [hunk 3]
```diff
@@ -32,3 +32,80 @@ Scenario: Install TestA from repository "test-1"
...
+ Then output line should "start" with "Installed groups:"
+ When I execute "dnf" command "install -y TestD" with "success"
+ Then package "TestD, TestE" should be "installed"
+ When I execute "dnf" command "group remove -y Testgroup" with "success"
+ Then package "TestA, TestB, TestC" should be "removed"
+ When I execute "dnf" command "group list Testgroup" with "success"
+ Then output line should "not start" with "Installed groups:"
+ When I execute "dnf" command "remove -y TestD, TestE" with "success"
+ Then package "TestD, TestE" should be "removed"
+
```
Comment (jsilhan) [inline dnf-docker-test/features/install_remove-1.feature]: ^ same here

File: dnf-docker-test/features/steps/test_behave.py [hunk 4]
```diff
@@ -46,13 +46,14 @@ def get_rpm_package_list():
 def get_rpm_package_version_list():
     """ Gets all installed packages in the system with version"""
     pkgverstr = subprocess.check_output(['rpm', '-qa', '--queryformat', '%{NAME}-%{VERSION}-%{RELEASE}\n'])
-    return pkgverstr.splitlines()
-
-
-def get_dnf_package_version_list():
```
Comment (jsilhan) [inline dnf-docker-test/features/steps/test_behave.py]: :thumbsup: good code reduction

File: dnf-docker-test/features/steps/test_behave.py [hunk 5]
```diff
@@ -156,20 +155,24 @@ def when_action_package(context, action, pkgs, manager):
...
     context.pre_rpm_packages_version = get_rpm_package_version_list()
     assert context.pre_rpm_packages_version
-    context.pre_dnf_packages_version = get_dnf_package_version_list()
-    assert context.pre_dnf_packages_version
     dnf_command_version = context.config.userdata['dnf_command_version']
     assert dnf_command_version
-    dnf_command_version = dnf_command_version + " " + command
+    if type_of_command == 'dnf':
+        dnf_command_version = dnf_command_version + " " + command
+    elif type_of_command == 'command line':
```
Comment (jsilhan) [inline dnf-docker-test/features/steps/test_behave.py]: "bash" instead?

File: dnf-docker-test/features/steps/test_behave.py [hunk 6]
```diff
@@ -253,3 +248,17 @@ def then_package_state(context, pkgs, state):
...
+    for line in context.cmd_output.split('\n'):
+        if line.startswith(line_start):
+            counter += 1
+    if state == 'start':
+        assert counter > 0
+    elif state == 'not start':
+        assert counter == 0
+    else:
+        raise AssertionError('The state {} is not allowed option for Then statement (allowed start, not start'
+                             .format(state))
```
Comment (jsilhan) [inline dnf-docker-test/features/steps/test_behave.py]: add empty line at the end of file, please. That's a UNIX convention. You should set automatically adding empty line at the end of file in your text editor.

File: dnf-docker-test/initial_settings/initial.py [hunk 7]
```diff
@@ -1,41 +1,55 @@
...
+        command_in_list = ['dnf', 'install', '--disablerepo=*', '--enablerepo=dnf-nightly', '--allowerasing', '-y',
+                           'dnf-plugins-core']
+        self.command_cl_runner(command_in_list)
+        command_in_list = ['dnf', 'upgrade', '-y', '--disablerepo=*', '--enablerepo=dnf-nightly', '--best']
+        self.command_cl_runner(command_in_list)
 
-    def upgrade_dnf_from_pull_request(self, pkg):
-        return subprocess.check_call(['dnf', 'install', '-y', '--disablerepo=*',
-                                      '--enablerepo=dnf-pull-requests', '--allowerasing', pkg])
+    def upgrade_copr_buit_package(self, pkg):
```
Comment (jsilhan) [inline dnf-docker-test/initial_settings/initial.py]: is *bui(l)t* a typo?

File: dnf-docker-test/README.md [hunk 8]
```diff
@@ -37,6 +37,10 @@ To rebuild the Docker image you can use the following command:
 $ cd dnf_docker_test/
 $ sudo docker build -t jmracek/dnftest:1.0.2 .
 ```
+To rebuild test-1 repository go into test-1 directory with rpm files and comps-f23.xml file and use following command:
+```
+$ createrepo -g comps-f23.xml .
```
Comment (jsilhan) [inline dnf-docker-test/README.md]: wouldn't it be better to call this from `dnf-docker-test/repo_create/test-1`?

Comment (j-mracek) [inline dnf-docker-test/README.md]: I have a plan to change repo creator when tests for group upgrade will be introduced.

File: dnf-docker-test/features/install_remove-1.feature (hunk 2 above)
Comment (j-mracek) [inline dnf-docker-test/features/install_remove-1.feature]: Added + more (test if Available and not Installed)

File: dnf-docker-test/features/steps/test_behave.py (hunk 6 above)
Comment (j-mracek) [inline dnf-docker-test/features/steps/test_behave.py]: Corrected, and I will try to setup. So farm I see warning, but sometimes I can overlook it.

**Comment by j-mracek** [comment]
New version is here. I believe, that it contains all requested changes. Thanks for advises. It helps.

**Comment by jsilhan** [comment]
Merged., thanks.

**Pull request merged**